### PR TITLE
Create util functions

### PR DESCRIPTION
**Pull request by iferencik**
Thank you for submitting a pull request!

## Description

Please describe what you changed briefly.

### Type of Pull Request

- [ ] Adding a feature
- [ ] Fixing a bug
- [ ] Maintaining documents
- [x] Others ()

### Verify the followings

- [ ] Code is up-to-date with the `develop` branch
- [ ] No build errors after `pnpm build`
- [ ] No lint errors after `pnpm lint`
- [ ] Run the tests with pnpm test and lint the project with pnpm lint and pnpm check
- [ ] Make sure all the existing features working well

### Changesets

- [ ] If your PR makes a change under `packages` folder that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

Refer to [CONTRIBUTING.MD](https://github.com/UNDP-Data/geohub/blob/develop/CONTRIBUTING.md) for more information.

### CODE REVIEW

File: sites/geohub/src/lib/helper/getValueFromRasterTileUrl.ts [hunk 1]
```diff
@@ -36,3 +36,31 @@ export const getValueFromRasterTileUrl = (
...
+    const values = value.split(',')
+    value = values.map((v) => Number(v))
+  } else if (value && paramName === 'colormap') {
+    if (Array.isArray(value)) {
+      // interval
+      value = JSON.parse(value) as number[][][]
+    } else {
+      // unique
+      value = JSON.parse(value) as unknown as { [key: string]: number[] }
+    }
```
Comment (JinIgarashi) [inline sites/geohub/src/lib/helper/getValueFromRasterTileUrl.ts:63]: Is it possible to call `getValueFromRasterTileUrl` function here? looks like same source code

File: sites/geohub/src/lib/helper/index.ts [hunk 2]
```diff
@@ -22,4 +22,5 @@ export * from './updateParamsInURL'
 export * from './sleep'
 export * from './getMaxValueOfCharsInIntervals'
 export * from './distinct'
+export * from './loadMap'
```
Comment (JinIgarashi) [inline sites/geohub/src/lib/helper/index.ts:25]: You can also add `getValueFromRasterTileUrlAsync.ts` to `index.ts`

File: sites/geohub/src/lib/types/Layer.ts [hunk 3]
```diff
@@ -9,4 +10,6 @@ export interface Layer {
   children?: Layer[]
   parentId?: string
   dataset: StacItemFeature
+  colormapName?: string
+  classificationMethod?: ClassificationMethodTypes
```
Comment (JinIgarashi) [inline sites/geohub/src/lib/types/Layer.ts:14]: You need to change to use this property instead of using map.fire.

But you maybe can do this change in another PR.

**Review by JinIgarashi — CHANGES_REQUESTED** [review]
Thanks, looks good.

File: sites/geohub/src/lib/types/Layer.ts (hunk 3 above)
Comment (iferencik) [reply sites/geohub/src/lib/types/Layer.ts:14]: I plan to remove those fire events later, once the refactoring is done

File: sites/geohub/src/lib/helper/getValueFromRasterTileUrl.ts (hunk 1 above)
Comment (iferencik) [reply sites/geohub/src/lib/helper/getValueFromRasterTileUrl.ts:63]: it is not exactly the same code, it is quite close. and unfortunately it is not possible to mix async and sync while preserting execution order. It is sad bu I had no choice

File: sites/geohub/src/lib/helper/index.ts (hunk 2 above)
Comment (iferencik) [reply sites/geohub/src/lib/helper/index.ts:25]: it is in the same module with getValueFromRasterTileUrl, so no need to add it

**Review by JinIgarashi — APPROVED** [review]
